### ISSUE TITLE
Possible lint for when range contains is being done against type bounds

### ISSUE THREAD

**Issue by ilovehans10** [rust-lang/rust-clippy#12926]
### What it does

This lint would change `(0u8..X).contains(&Y)` to `Y < X` in cases where 0u8 can be replaced with the lower bound for the type. It would also do the same for the upper bound for the type, e.g. `(X..u8::max_value()).contains(Y)` into `Y > X`.

There is also the matter of `(u8::min_value()..=u8::max_value()).contains(&5)` doesn't throw any sort of warning about checking the full range of all values for a value. Should I open another issue about adding a lint for that too?

### Advantage

Clarifies that range is being checked all the way to the type bounds.

Intuitive corollary to [manual_range_contains](https://rust-lang.github.io/rust-clippy/master/#/manual_range_contains).

### Drawbacks

To my knowledge, in order to implement range requires implementing PartialOrd, but implementing PartialOrd doesn't include any surefire way to check the bounds of the type. However, my testing seems to show that using single sided ranges (..X) works for types that implement their own PartialOrd, so I must just be missing something.

I'm unsure where this should be listed, but there is also a borrow that doesn't have to be done if the comparison is being done instead of the contains check.

### Example

```rust
assert!((0..5).contains(&2u8));
assert!((0..=5).contains(&5u8));
```

Could be written as:

```rust
assert!(2u8 < 5);
assert!(5u8 <= 5);
```
